subprojects/core/src/main/java/org/gradle/internal/classpath/Instrumented.java

```diff
@@ -477,7 +477,7 @@ public static String kotlinIoFilesKtReadTextDefault(File receiver, Charset chars
         lazyKotlinStaticDefaultHandle(FilesKt.class, "readText", String.class, File.class, Charset.class);
 
     public static String filesReadString(Path file, String consumer) throws Throwable {
-        listener().fileOpened(file.toFile(), consumer);
+        FileUtils.tryReportFileOpened(file, consumer);
```

Comment:
Good catch!

Reply:
shout out to [@netvl](https://github.com/netvl), author of https://github.com/gradle/gradle/issues/25044!